packages/@aws-cdk/aws-lambda/lib/code.ts

```diff
@@ -54,6 +54,15 @@ export abstract class Code {
     return new AssetCode(path, options);
   }
 
+  /**
+   * Loads the function code from an Asset construct.
+   *
+   * @param asset An Asset construct containing the Lambda code bundle or a .zip file
+   */
+  public static fromAssetConstruct(asset: s3_assets.Asset): AssetConstructCode {
```

Comment:
```suggestion
  public static fromS3Asset(asset: s3_assets.Asset): S3AssetCode {
```

What about this naming instead since it only accepts an S3 asset? I could also imagine there being a `fromDockerAsset(asset: ecr_assets.DockerImageAsset)`